### ISSUE TITLE
AttributeError: '_MigrateConfig' object has no attribute 'configure_args'

### ISSUE THREAD

**Issue by krinkere** [miguelgrinberg/flasky#174]
Created brand new environment

git checkout 9a
pip install -r requirements.txt
python manage.py db upgrade

Got 
AttributeError: '_MigrateConfig' object has no attribute 'configure_args'

I saw that it might be because of Flask-Migrate
but I checked that the one in requirements and what i have in my virtual environment do match

**Comment by miguelgrinberg** [comment]
This is a weird error. The `configure_args` attribute in Flask-Migrate was introduced in release 1.4.0. The requirements file in this repo points to 1.1.0, which predates the work that introduced this attribute. My guess is that you have a mixed new and old releases.

Maybe you generated your migration repository with a newer release of Flask-Migrate, and then downgraded it? That would make the repository incompatible with the older Flask-Migrate, the migration repository needs to be generated with the same version you use for the application or an older release, but not a newer release.

**Comment by krinkere** [comment]
It is weird indeed. What I did to get around it - I removed specified versions in requirements file and let pip download the latest (with fingers crossed that it won't have incompatibility issues between different packages)

Just tried to recreate the issue with same results
I created different env with 
conda create -n flasky python=2.7
source activate flasky
pip install -r requirements/dev.txt

(flasky) $ conda list
# packages in environment at /ofrconda/envs/flasky:
#
alembic                   0.6.2                     <pip>
bleach                    1.4                       <pip>
blinker                   1.3                       <pip>
colorama                  0.2.7                     <pip>
Flask                     0.10.1                    <pip>
Flask-Bootstrap           3.0.3.1                   <pip>
Flask-HTTPAuth            2.7.0                     <pip>
Flask-Login               0.3.1                     <pip>
Flask-Mail                0.9.0                     <pip>
Flask-Migrate             1.1.0                     <pip>
Flask-Moment              0.2.1                     <pip>
Flask-PageDown            0.1.4                     <pip>
Flask-Script              0.6.6                     <pip>
Flask-SQLAlchemy          1.0                       <pip>
Flask-WTF                 0.9.4                     <pip>
ForgeryPy                 0.1                       <pip>
html5lib                  1.0b3                     <pip>
openssl                   1.0.2h                        1    
pip                       8.1.2                    py27_0    
python                    2.7.12                        1    
readline                  6.2                           2   
setuptools                23.0.0                   py27_0   
sqlite                    3.13.0                        0   
tk                        8.5.18                        0    
wheel                     0.29.0                   py27_0   
zlib                      1.2.8                         3   
httpie                    0.7.2                     <pip>
itsdangerous              0.23                      <pip>
Jinja2                    2.7.1                     <pip>
Mako                      0.9.1                     <pip>
Markdown                  2.3.1                     <pip>
MarkupSafe                0.18                      <pip>
Pygments                  1.6                       <pip>
requests                  2.1.0                     <pip>
six                       1.4.1                     <pip>
SQLAlchemy                0.9.9                     <pip>
Werkzeug                  0.10.4                    <pip>
WTForms                   1.0.5                     <pip>

**Comment by miguelgrinberg** [comment]
Do you have the full stack trace of the original error?

**Comment by krinkere** [comment]
(flasky) $ python manage.py db upgrade
Traceback (most recent call last):
  File "manage.py", line 29, in <module>
    manager.run()
  File "/conda/envs/flasky/lib/python2.7/site-packages/flask_script/__init__.py", line 405, in run
    result = self.handle(sys.argv[0], sys.argv[1:])
  File "/conda/envs/flasky/lib/python2.7/site-packages/flask_script/__init__.py", line 384, in handle
    return handle(app, *positional_args, **kwargs)
  File "/conda/envs/flasky/lib/python2.7/site-packages/flask_script/commands.py", line 145, in handle
    return self.run(*args, **kwargs)
  File "/conda/envs/flasky/lib/python2.7/site-packages/flask_migrate/__init__.py", line 98, in upgrade
    command.upgrade(config, revision, sql = sql, tag = tag)
  File "/conda/envs/flasky/lib/python2.7/site-packages/alembic/command.py", line 124, in upgrade
    script.run_env()
  File "/conda/envs/flasky/lib/python2.7/site-packages/alembic/script.py", line 199, in run_env
    util.load_python_file(self.dir, 'env.py')
  File "/conda/envs/flasky/lib/python2.7/site-packages/alembic/util.py", line 199, in load_python_file
    module = load_module(module_id, path)
  File "/conda/envs/flasky/lib/python2.7/site-packages/alembic/compat.py", line 55, in load_module
    mod = imp.load_source(module_id, path, fp)
  File "migrations/env.py", line 87, in <module>
    run_migrations_online()
  File "migrations/env.py", line 76, in run_migrations_online
    **current_app.extensions['migrate'].configure_args)
AttributeError: '_MigrateConfig' object has no attribute 'configure_args'

**Comment by miguelgrinberg** [comment]
Great. So take a look at your `migrations/env.py` file. That is clearly a env.py generated by a newer version of Flask-Migrate. If you compare it against the version of that file in this repository ([link](https://github.com/miguelgrinberg/flasky/blob/master/migrations/env.py)) you will see that it is different, so you generated that file on your own instead of taking the version that is compatible with the dependencies in this repository.

**Comment by krinkere** [comment]
Indeed they are different... and now I know what happened. And it all makes sense now. Duh... Thank you for pointing me to the env.py file which made it snap in my head.

I got the source code and I was running it against my default environment where I had Flask-Migrate 2.0,0 installed. I was running examples just fine and then when the book (which is awesome btw) required more packages to be installed that i did not have, i decided to create new environment not to mess up my current one.

but... since i was running upgrade/etc scripts, they changed env.py script - hence when I created new env with Flask-Migrate 1.1.0 and still ran within the same directory of the source code, it broke.

All makes sense now. After hard git reset everything is working just fine. Thanks again for troubleshooting it!
